spdmlib/src/message/capability.rs

```diff
@@ -102,10 +118,17 @@ impl SpdmCodec for SpdmGetCapabilitiesRequestPayload {
                 {
                     return None;
                 }
-            } else if flags.contains(SpdmRequestCapabilityFlags::CHAL_CAP)
-                || flags.contains(SpdmRequestCapabilityFlags::MUT_AUTH_CAP)
-            {
-                return None;
+            } else {
+                if flags.contains(SpdmRequestCapabilityFlags::CHAL_CAP)
+                    || flags.contains(SpdmRequestCapabilityFlags::MUT_AUTH_CAP)
+                {
+                    return None;
+                }
+                if context.negotiate_info.spdm_version_sel >= SpdmVersion::SpdmVersion13
```

Comment:
Capabilities Flags checks are aligned with spdm spec 1.3 and libspdm implementation.